src/plugins/discover/public/application/angular/discover.js

```diff
@@ -739,57 +530,6 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
     });
   });
 
-  async function saveDataSource(saveOptions) {
-    await $scope.updateDataSource();
-
-    savedSearch.columns = $scope.state.columns;
-    savedSearch.sort = $scope.state.sort;
-
-    try {
-      const id = await savedSearch.save(saveOptions);
-      $scope.$evalAsync(() => {
-        if (id) {
-          toastNotifications.addSuccess({
-            title: i18n.translate('discover.notifications.savedSearchTitle', {
-              defaultMessage: `Search '{savedSearchTitle}' was saved`,
-              values: {
-                savedSearchTitle: savedSearch.title,
-              },
-            }),
-            'data-test-subj': 'saveSearchSuccess',
-          });
-
-          if (savedSearch.id !== $route.current.params.id) {
-            history.push(`/view/${encodeURIComponent(savedSearch.id)}`);
-          } else {
-            // Update defaults so that "reload saved query" functions correctly
-            setAppState(getStateDefaults());
-            chrome.docTitle.change(savedSearch.lastSavedTitle);
-            chrome.setBreadcrumbs([
-              {
-                text: discoverBreadcrumbsTitle,
-                href: '#/',
-              },
-              { text: savedSearch.title },
-            ]);
-          }
-        }
-      });
-      return { id };
-    } catch (saveError) {
-      toastNotifications.addDanger({
-        title: i18n.translate('discover.notifications.notSavedSearchTitle', {
-          defaultMessage: `Search '{savedSearchTitle}' was not saved.`,
-          values: {
-            savedSearchTitle: savedSearch.title,
-          },
-        }),
-        text: saveError.message,
-      });
-      return { error: saveError };
-    }
-  }
-
```

Comment:
Extracted and refactored, to 
https://github.com/elastic/kibana/blob/5a1f3ef6f740315b7b41013f840a36b48cd68c66/src/plugins/discover/public/application/helpers/persist_saved_search.ts
think this is the biggest change, just moved here without changing logic is a bit of an understatement